Brokerages/GDAX/GDAXBrokerage.Messaging.cs

```diff
@@ -0,0 +1,406 @@
+                            Time = DateTime.UtcNow,
+                            Symbol = item,
+                            TickType = TickType.Quote
+                            //todo: tick volume                          
+                        };
+
+                        this.Ticks.Add(updating);
+                    }
+                }
+            }
+
+            var products = ChannelList.Select(s => s.Value.Symbol.Substring(0, 3) + "-" + s.Value.Symbol.Substring(3)).ToArray();
+
+            var payload = new
+            {
+                type = "subscribe",
+                product_ids = products,
+                channels = _channelNames
+            };
+
+            if (payload.product_ids.Length == 0)
+            {
+                return;
+            }
+
+            var token = GetAuthenticationToken(JsonConvert.SerializeObject(payload), "GET", "/users/self/verify");
+
+            var json = JsonConvert.SerializeObject(new
+            {
+                type = payload.type,
+                channels = payload.channels,
+                product_ids = payload.product_ids,
+                SignHeader = token.Signature,
+                KeyHeader = ApiKey,
+                PassHeader = _passPhrase,
+                TimeHeader = token.Timestamp
+            });
+
+            WebSocket.Send(json);
+
+
+            Log.Trace("Subscribe: Sent subcribe.");
+
+        }
+
+        /// <summary>
+        /// Poll for new tick to refresh conversion rate of non-USD denomination
+        /// </summary>
+        /// <param name="symbol"></param>
+        /// <param name="client"></param>
+        public void PollTick(Symbol symbol)
+        {
+
+            int delay = 36000000;
+            var token = _canceller.Token;
+            var listener = Task.Factory.StartNew(() =>
+            {
+                Log.Trace("PollLatestTick: " + "Started polling for ticks: " + symbol.Value.ToString());
+                while (true)
+                {
+                    var rate = GetConversionRate(symbol.Value.Replace("USD", ""));
+
+                    lock (_tickLocker)
+                    {
+                        var latest = new Tick
+                        {
+                            Value = rate,
+                            Time = DateTime.UtcNow,
+                            Symbol = symbol
+                        };
+                        Ticks.Add(latest);
+                    }
+
+                    Thread.Sleep(delay);
+                    if (token.IsCancellationRequested) break;
+                }
+                Log.Trace("PollLatestTick: " + "Stopped polling for ticks: " + symbol.Value.ToString());
+            }, token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
+        }
+
+        private decimal GetConversionRate(string currency)
+        {
+            var response = RateClient.Execute(new RestSharp.RestRequest(Method.GET));
```

Comment:
this is a no-op right? Assume currently only converts USD balances?

Reply:
Yes, this workarounds the conversion rate not found error for non USD denominated currencies. The alternative was to require an FXCM or Oanda subscription or use the compound rate from two tickers. This issue might deserve a resolution that goes beyond one brokerage.